src/test/java/space/obminyashka/items_exchange/controller/UserControllerIntegrationTest.java

```diff
@@ -174,14 +174,23 @@ void updateUserPassword_WhenPasswordConfirmationWrong_ShouldThrowIllegalArgument
         assertTrue(message.contains(getMessageSource(ResponseMessagesHandler.ValidationMessage.DIFFERENT_PASSWORDS)));
     }
 
-    @Test
+    @ParameterizedTest
     @WithMockUser(username = "user")
-    void updateUserEmail_WhenEmailConfirmationWrong_ShouldThrowIllegalArgumentException() throws Exception {
-        updateUserEmailBasicTest(NEW_VALID_EMAIL, NEW_INVALID_DUPLICATE_EMAIL, status().isBadRequest(), "invalid.confirm.email");
+    @MethodSource("listInvalidEmail")
+    void updateUserEmail_WhenEmailConfirmationWrong_ShouldThrowIllegalArgumentException(String email) throws Exception {
+        UserChangeEmailDto userChangeEmailDto = new UserChangeEmailDto(email);
+        sendDtoAndGetMvcResult(put(USER_SERVICE_CHANGE_EMAIL), userChangeEmailDto, status().isBadRequest());
+    }
+
+    private static Stream<Arguments> listInvalidEmail() {
+        return Stream.of(
+                Arguments.of(NEW_INVALID_EMAIL),
+                Arguments.of(INVALID_EMAIL)
+        );
     }
 
-    private void updateUserEmailBasicTest(String email, String confirmationEmail, ResultMatcher expectedStatus, String expectedExceptionKey) throws Exception {
-        UserChangeEmailDto userChangeEmailDto = new UserChangeEmailDto(email, confirmationEmail);
+    private void updateUserEmailBasicTest(String email, ResultMatcher expectedStatus, String expectedExceptionKey) throws Exception {
```

Comment:
It would be much better to merge **updateUserEmail_WhenUserEnteredOldEmail_ShouldThrowDataConflictException** and **updateUserEmail_WhenUserEnteredExistedEmail_ShouldThrowDataConflictException** to a new parametrized test instead of updating a method used by them both.

Reply:
Done